adr/context.py

```diff
@@ -0,0 +1,287 @@
+import inspect
+from jsone.interpreter import ExpressionEvaluator
+from jsone.prattparser import prefix
+
+COMMON_CONTEXTS = [
+    [['-B', '--branch'],
+     {'default': 'mozilla-central',
+      'help': "Branches to query results from",
+      }],
+    [['-b', '--build-type'],
+     {'default': 'opt',
+      'help': "Build type (default: opt)",
+      }],
+    [['--from'],
+     {'dest': 'from_date',
+      'default': 'today-week',
+      'help': "Starting date to pull data from, defaults "
+              "to a week ago",
+      }],
+    [['--to'],
+     {'dest': 'to_date',
+      'default': 'eod',  # end of day
+      'help': "Ending date to pull data from, defaults "
+              "to now",
+      }],
+    [['--path'],
+     {'required': True,
+      'default': 'dom/indexedDB',
+      'help': "Path relative to repository root (file or directory)",
+      }],
+    [['-p', '--platform'],
+     {'default': 'windows10-64',
+      'help': "Platform to limit results to (default: windows10-64)",
+      }],
+    [['-r', '--revision'],
+     {'dest': 'rev',
+      'required': True,
+      'default': '5b33b070378a',
+      'help': "Revision to limit results to",
+      }],
+    [['-t', '--test'],
+     {'required': True,
+      'dest': 'test_name',
+      'help': "Path to a test file",
+      }],
+
+    [['--limit'],
+     {'type': int,
+      'default': 10,
+      'help': "limit the number of rows in result"
+      }],
+
+    [['--format'],
+     {'type': str,
+      'default': 'table',
+      'help': "format of result"
+      }],
+    [['--sort-key'],
+     {'type': int,
+      'dest': 'sort_key',
+      'default': 4,
+      'help': "Key to sort on (int, 0-based index)",
+      }]
+]
+"""
+These are commonly used arguments which can be re-used. They are shared to
+provide a consistent CLI across recipes/queries.
+"""
+
+
+class ZeroContextEvaluator(ExpressionEvaluator):
+    """
+    Zero context evaluator is used in conjunction with extract_context_names
+    to extract argument context as a token value
+    """
+    @staticmethod
+    def update_missing(dest_dict, source_dict):
+        """
+        Update missing key from source dict to destination dict
+        :param dict dest_dict: destination dist to be updated
+        :param dict source_dict: source dict used to update destination
+        :return: None
+        """
+        for k, v in source_dict.items():
+            if k not in dest_dict:
+                dest_dict[k] = v
+
+    def __init__(self):
+        # Init evaluator with fake empty context
+        super(ZeroContextEvaluator, self).__init__({})
+        # update infix_rules and prefix_rules with override function
+        ZeroContextEvaluator.update_missing(self.infix_rules,
+                                            super(ZeroContextEvaluator, self).infix_rules)
+        ZeroContextEvaluator.update_missing(self.prefix_rules,
+                                            super(ZeroContextEvaluator, self).prefix_rules)
```

Comment:
I'm a bit confused about why this is necessary. Shouldn't `self.infix_rules` be the same as `PrattParser.infix_rules` at this point?

Reply:
Yes, I updated the code to make it clearer.